src/mlpack/core/util/to_lower.hpp

```diff
@@ -0,0 +1,36 @@
+/**
+ * @file to_lower.hpp
+ * @author Himanshu Pathak
+ *
+ * Convert a string to lower-case.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_UTIL_LOWER_STRING_HPP
+#define MLPACK_CORE_UTIL_LOWER_STRING_HPP
+
+namespace mlpack {
+namespace util {
+
+/**
+ * ToLower convert a string into a string of lower case characters
+ * only.
+ *
+ * @param str String to convert string.
+ */
```

Comment:
May suggest a little neatening of the docstring here.
```suggestion
/**
 * Convert a string to lowercase letters.
 *
 * @param str The string to convert.
 */
```